lib/posix/pthread.c

```diff
@@ -145,6 +146,17 @@ int pthread_create(pthread_t *newthread, const pthread_attr_t *attr,
 		return EINVAL;
 	}
 
+	pthread_mutex_lock(&pthread_pool_lock);
+	for (i = 0; i < CONFIG_MAX_PTHREAD_COUNT; i++) {
+		thread = &posix_thread_pool[i];
+		if (thread->state == PTHREAD_TERMINATED) {
+			thread->state = PTHREAD_JOINABLE;
+			break;
+		}
+	}
+	pthread_num = i;
```

Comment:
I'm sorry for nitpicking, but why do we need both of "i" and "pthread_num", if you end up unconditionally assigning one to another? Just use one var in the first place, appropriately named ("i" or "pthread_no" are examples of good names IMHO).